test/e2e/logs_test.go

```diff
@@ -0,0 +1,47 @@
+package e2e
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// testNodeLogs ensures that all required log files were created, and copies them to the test's artifact directory
+func testNodeLogs(t *testing.T) {
+	// All these paths are relative to /var/log/
+	logFiles := []string{
+		"kube-proxy\\kube-proxy.exe.INFO",
+		"kube-proxy\\kube-proxy.exe.ERROR",
+		"kube-proxy\\kube-proxy.exe.WARNING",
+		"hybrid-overlay\\hybridOverlay.log",
+		"kubelet\\kubelet.log",
+	}
+
+	nodeArtifacts := filepath.Join(os.Getenv("ARTIFACT_DIR"), "nodes")
+	for _, node := range gc.nodes {
+		nodeDir := filepath.Join(nodeArtifacts, node.Name)
+		for _, file := range logFiles {
+			// Use oc to read the expected log files
+			cmd := exec.Command("oc", "adm", "node-logs", "--path", file, node.Name)
```

Comment:
Can't this be a API call?

Reply:
It can, but since the ticket is to make `oc adm node-logs` work, I thought it would be best to test with what we need to be compatible with.

I can switch to using the API call, if that is what we really care about supporting.

Reply:
That is a good point. Please add a comment stating that and it should be good.

Reply:
nit: I'd prefer having this in a shell script considering we're not making api calls and technically this is not validation but it is log-collection.

Reply:
It is a validation that we are able to collect the logs